static-analysis/teammates-macker.xml

```diff
@@ -1,45 +1,338 @@
+                            <include class="teammates.client.**" />
+                        </from>
+                    </allow>
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="TestDriverComponent: Only certain test cases can test storage:entity">
+            <access-rule>
+                <deny>
+                    <from class="teammates.test.**" />
+                    <to class="teammates.storage.entity.**" />
+                    <allow>
+                        <from class="teammates.test.cases.datatransfer.**" />
+                    </allow>
+                    
+                    <!-- Violations that need fixing -->
+                    <allow>
+                        <from class="teammates.test.cases.logic.StudentsLogicTest" />
+                        <to class="teammates.storage.entity.CourseStudent" />
+                    </allow>
+                    <allow>
+                        <from class="teammates.test.cases.storage.FeedbackResponseCommentsDbTest" />
+                        <to class="teammates.storage.entity.FeedbackResponseComment" />
+                    </allow>
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="TestDriverComponent: Only certain test cases can access GaeSimulation/BackDoorLogic">
+            <access-rule>
+                <deny>
+                    <to>
+                        <include class="teammates.test.driver.GaeSimulation" />
+                        <include class="teammates.logic.backdoor.BackDoorLogic" />
+                    </to>
+                    <allow>
+                        <from>
+                            <include class="teammates.test.cases.storage.**" />
+                            <include class="teammates.test.cases.logic.**" />
+                            <include class="teammates.test.cases.action.**" />
+                            <include class="teammates.test.cases.automated.**" />
+                            <include class="teammates.test.cases.search.**" />
+                            <include class="teammates.test.cases.BaseComponentTestCase" />
+                            
+                            <!-- Violation that needs fixing -->
+                            <include class="teammates.test.cases.pagedata.StudentFeedbackResultsPageDataTest" />
+                        </from>
+                    </allow>
+                    <allow>
+                        <!-- to import sample data for new instructor -->
+                        <from class="teammates.ui.controller.AdminInstructorAccountAddAction" />
+                        <to class="teammates.logic.backdoor.BackDoorLogic" />
+                    </allow>
+                    <allow>
+                        <from class="teammates.logic.backdoor.BackDoorServlet" />
+                        <to class="teammates.logic.backdoor.BackDoorLogic" />
```

Comment:
If this ruleset is about test code, the above two `allow`s should not be relevant.